sounddevice.py

```diff
@@ -74,10 +74,17 @@
     if _platform.system() == 'Darwin':
         _libname = 'libportaudio.dylib'
     elif _platform.system() == 'Windows':
-        if 'SD_ENABLE_ASIO' in _os.environ:
-            _libname = 'libportaudio' + _platform.architecture()[0] + '-asio.dll'
+        machine = _platform.machine().lower()
+        if machine in ('arm64', 'aarch64'):
+            if 'SD_ENABLE_ASIO' in _os.environ:
+                _libname = 'libportaudioarm64-asio.dll'
+            else:
+                _libname = 'libportaudioarm64.dll'
         else:
-            _libname = 'libportaudio' + _platform.architecture()[0] + '.dll'
+            if 'SD_ENABLE_ASIO' in _os.environ:
+                _libname = 'libportaudio' + _platform.architecture()[0] + '-asio.dll'
+            else:
+                _libname = 'libportaudio' + _platform.architecture()[0] + '.dll'
```

Comment:
I think the logic is fine, but un-nesting the `if` clauses would probably make it a bit more compact?

```python
        if _platform.machine().lower() in ('arm64', 'aarch64'):
            platform_suffix = 'arm64'
        else:
            platform_suffix = _platform.architecture()[0]
        if 'SD_ENABLE_ASIO' in _os.environ:
            _libname = 'libportaudio' + platform_suffix + '-asio.dll'
        else:
            _libname = 'libportaudio' + platform_suffix + '.dll'
```